### PR TITLE
ENG-1167 - Endpoint to delete service and associated audiences

### PR DESCRIPTION
**Pull request by blinktag**

### CODE REVIEW

File: apis/grpcapi/external.go [hunk 1]
```diff
@@ -737,6 +710,95 @@ func (s *ExternalServer) DeleteAudience(ctx context.Context, req *protos.DeleteA
...
+		deleteReq := &protos.DeleteAudienceRequest{
+			Audience: audience,
+			Force:    util.BoolPtr(true),
+		}
+
+		if err := s.Options.StoreService.DeleteAudience(ctx, deleteReq); err != nil {
+			return util.StandardResponse(ctx, protos.ResponseCode_RESPONSE_CODE_INTERNAL_SERVER_ERROR, err.Error()), nil
+		}
+
+		// Broadcast delete to other nodes so that they can emit an event for GetAllStream()
```
Comment (dselans) [inline apis/grpcapi/external.go:789]: 👍 great comment

File: go.mod [hunk 2]
```diff
@@ -11,7 +11,7 @@ require (
 	github.com/patrickmn/go-cache v2.1.0+incompatible // TODO: Switch this to bigcache
 	github.com/pkg/errors v0.9.1
 	github.com/sirupsen/logrus v1.9.3
-	github.com/streamdal/snitch-protos v0.0.101
+	github.com/streamdal/snitch-protos v0.0.106-0.20231010211049-4e65e47b0eb9
```
Comment (dselans) [inline go.mod:14]: Is this referencing the right proto version? Is this public/available?

File: services/store/store.go [hunk 3]
```diff
@@ -918,6 +920,27 @@ func (s *Store) GetAudiences(ctx context.Context) ([]*protos.Audience, error) {
 	return audiences, nil
 }
 
+func (s *Store) GetAudiencesByService(ctx context.Context, serviceName string) ([]*protos.Audience, error) {
```
Comment (dselans) [inline services/store/store.go:923]: Def need to add tests for these helper methods at some point - a lot of stuff depends on this working correctly. Same as `util.AudienceTo*` and `util.AudienceFrom*`

**Review by dselans — APPROVED** [review]
Couple of minor questions and comments.

File: go.mod (hunk 2 above)
Comment (blinktag) [reply go.mod:14]: It's currently referencing a branch since there are multiple proto changes waiting to go out

**Pull request merged**